crml/transaction-payment/src/lib.rs

```diff
@@ -277,7 +288,9 @@ where
 				ExistenceRequirement::KeepAlive,
 			) {
 				Ok(imbalance) => imbalance,
-				Err(_) => return Err(TransactionValidityError::Invalid(InvalidTransaction::Payment)),
+				Err(_) => {
+					return Err(InvalidTransaction::Custom(error_code::INSUFFICIENT_SELLER_CORE_ASSET_BALANCE).into())
```

Comment:
This error is abit confusing, it really means the user could not pay the transaction fee so would be good to have a simple, more meaningful error e.g.

"error_code::INSUFFICIENT_FEE_ASSET"